### PR TITLE
This PR migrates Ditto to Apache Pekko

### PR DESCRIPTION
**Pull request by alstanchev**
Since the change of akka's license to "Business Source License (BSL) v1.1" this PR:

1. Migrates all the code to use Apache Pekko framework (fork from Akka 2.6.21)
2. Adapts the documentation, third party notices
3. And all necessary changes.

Resolves: #1477

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -37,24 +39,30 @@
...
+        <pekko-persistence-inmemory.version>1.0.0</pekko-persistence-inmemory.version>
+        <pekko-management.version>1.0.0</pekko-management.version>
+        <pekko-connector-kafka.version>1.0.0</pekko-connector-kafka.version>
+        <parboiled.version>2.5.0</parboiled.version>
+
+        <!--  Necessary for pekko-persistence-mongodb -->
+        <metrics4-scala.version>4.2.9</metrics4-scala.version>
+
+        <!-- Keep these version consistent with pekko-persistence-mongodb.version's build.sbt -->
         <mongo-java-driver.version>4.3.4</mongo-java-driver.version>
```
Comment (thjaeckle) [inline bom/pom.xml:60]: @alstanchev could you upgrade to the Mongo driver version pekko-persistence-mongo uses?

**Review by thjaeckle — COMMENTED** [review]
One first finding when roughly scrolling over the changes.
Will have a closer look soon.

Great effort @alstanchev :+1:

File: bom/pom.xml (hunk 1 above)
Comment (thjaeckle) [reply bom/pom.xml:60]: Now we can also update to `pekko-persistence-mongodb` 1.0.1 :)

File: documentation/src/main/resources/pages/ditto/installation-operating.md [hunk 2]
```diff
@@ -16,6 +16,37 @@ This page shows the basics for operating Ditto.
 Ditto has many config parameters which can be set in the config files or via environment variables.
 This section will cover some of Ditto's config parameters.
 
+### Migrating to Ditto 3.4.x
```
Comment (thjaeckle) [inline documentation/src/main/resources/pages/ditto/installation-operating.md:19]: @alstanchev this section should be moved to a new `release_notes_340.md`.

We always documented the migration notes in the release notes.
At the current location, probably they are not found.

File: documentation/src/main/resources/pages/ditto/installation-operating.md [hunk 3]
```diff
@@ -16,6 +16,37 @@ This page shows the basics for operating Ditto.
...
+```
+
+Connectivity:
+```markdown
+* -Dpekko-contrib-mongodb-persistence-connection-journal.overrides.metadata-index=akka_persistence_metadata_pid
+* -Dpekko-contrib-mongodb-persistence-connection-journal-read.overrides.metadata-index=akka_persistence_metadata_pid
+* -Dpekko-contrib-mongodb-persistence-connection-remember-journal.overrides.metadata-index=akka_persistence_metadata_pid
+```
+
+And also a full cluster recreate is required, rolling update is not supported as there are changes in the management
```
Comment (thjaeckle) [inline documentation/src/main/resources/pages/ditto/installation-operating.md:45]: @alstanchev do the changes in management URLs require adjusting anything e.g. in the Helm chart?

**Review by thjaeckle — CHANGES_REQUESTED** [review]
@alstanchev I reviewed by also starting locally and (manually) testing if data would still be there with Pekko - seems good as far as I can tell 👍 

I had just one remark about the migration guide.

File: documentation/src/main/resources/pages/ditto/release_notes_340.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+This release is completely [IP (intellectual property) checked by the Eclipse Foundation](https://www.eclipse.org/projects/handbook/#ip)
+meaning that project code as well as all used dependencies were "[...] reviewed to ensure that the copyrights
+expressed are correct, licensing is valid and compatible, and that other issues have been uncovered and properly
+investigated."
+
+
+## Changelog
+
+Compared to the latest release [3.3.6](release_notes_336.html), no features were added.
```
Comment (thjaeckle) [inline documentation/src/main/resources/pages/ditto/release_notes_340.md:20]: This is not true .. E.g. the `documentDB` support is in 3.4.0, as well as other topics: https://github.com/eclipse-ditto/ditto/milestone/61

Comment (alstanchev) [reply documentation/src/main/resources/pages/ditto/release_notes_340.md:20]: Oh. My bad. I thought that it went in with the previous release.

**Comment by alstanchev** [comment]
@thjaeckle from i have tested this whit the helm chart and it is working. From my point of view it is ready for merge.

**Review by thjaeckle — APPROVED** [review]
FMPOV also good to merge :+1:

**Pull request merged**